server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -223,7 +225,8 @@ public DesiredBalance compute(
                         && routingAllocation.deciders()
                             .canAllocate(shardRouting, routingNode, routingAllocation)
                             .type() != Decision.Type.NO) {
-                        final var shardToInitialize = unassignedReplicaIterator.initialize(nodeId, null, 0L, changes);
+                        final long expectedShardSize = getExpectedShardSize(shardRouting, 0L, routingAllocation);
+                        final var shardToInitialize = unassignedReplicaIterator.initialize(nodeId, null, expectedShardSize, changes);
```

Comment:
Likewise, reverting this change doesn't make any `DesiredBalanceComputerTests` fail.

Reply:
Okay, this and below is no longer necessary. Now we rely on `getExpectedShardSize` rather then on expected shard size set on shard itself.